content/en/docs/collector/architecture.md

```diff
@@ -0,0 +1,368 @@
+    timeout: 10s
+
+service:
+  pipelines:
+    traces: # a pipeline of “traces” type
+      receivers: [zipkin]
+      processors: [batch]
+      exporters: [jaeger]
+    traces/2: # another pipeline of “traces” type
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [otlp]
+```
+
+When the Collector loads this config, the result looks like this diagram:
+
+```mermaid
+---
+title: Pipeline "traces"
+---
+flowchart LR
+    R1("`zipkin Receiver`") --> P1["`#quot;batch#quot; Processor`"]
+    P1 --> E1[["`#quot;jaeger#quot; Exporter`"]]
+
+    classDef default fill:#e3e8fc,stroke:#4f62ad;
+```
+
+```mermaid
+---
+title: Pipeline "traces/2"
+---
+flowchart LR
+    R1("`otlp Receiver`") --> P1["`#quot;batch#quot; Processor`"]
+    P1 --> E1[["`#quot;otlp#quot; Exporter`"]]
+
+    classDef default fill:#e3e8fc,stroke:#4f62ad;
+```
+
+Note that each `batch` processor is an independent instance, although they are
+configured the same way with a `send_batch_size` of `10000`.
+
+> The same name of the processor must not be referenced multiple times in the
+> `processors` key of a single pipeline.
+
+## <a name="opentelemetry-agent"></a>Running as an agent
+
+On a typical VM/container, user applications are running in some processes/pods
+with an OpenTelemetry library. Previously, the library did all the recording,
+collecting, sampling, and aggregation of traces, metrics, and logs, and then
+either exported the data to other persistent storage backends through the
+library exporters, or displayed it on local zpages. This pattern has several
+drawbacks, for example:
+
+1. For each OpenTelemetry library, exporters and zpages must be re-implemented
+   in native languages.
+2. In some programming languages (for example, Ruby or PHP), it is difficult to
+   do the stats aggregation in process.
+3. To enable exporting of OpenTelemetry spans, stats, or metrics, application
+   users need to manually add library exporters and redeploy their binaries.
+   This is especially difficult when an incident has occurred, and users want to
+   use OpenTelemetry to investigate the issue right away.
+4. Application users need to take the responsibility for configuring and
+   initializing exporters. These tasks are error-prone (for example, setting up
+   incorrect credentials or monitored resources), and users may be reluctant to
+   “pollute” their code with OpenTelemetry.
+
+To resolve the issues above, you can run OpenTelemetry Collector as an agent.
+The agent runs as a daemon in the VM/container and can be deployed independent
+of the library. Once the agent is deployed and running, it should be able to
+retrieve traces, metrics, and logs from the library, and export them to other
+backends. We may also give the agent the ability to push configurations (such as
+sampling probability) to the library. For those languages that cannot do stats
+aggregation in process, they can send raw measurements and have the agent do the
+aggregation.
+
+```mermaid
```

Comment:
May I have your thoughts on this Mermaid diagram? 

Mermaid's algorithm makes it surprisingly difficult to keep nodes in order. The workaround is subgraphs, but some may find the resulting boxes too messy for this diagram. Before I go on to the final diagram, which has a similar issue, I'd like to hear your thoughts. Thanks!

Reply:
I neutralized some of the subgraph formatting in 677af81. I think it looks better now but would still like to hear others' opinions.

Reply:
Considering Mermaid's limitations, the diagram looks great to me!